Shopify/cli
# Add new checkout UI template

### WHY are these changes introduced?

Fixes #671
Fixes #673

### WHAT is this pull request doing?

Adds a new option to `yarn generate extension`:

<img width="267" alt="Screen Shot 2022-10-26 at 3 04 34 PM" src="https://user-images.githubusercontent.com/690791/198114297-48184f7f-1840-4838-a472-f28202dc84d3.png">

This option should only be available to shopifolk.  Selecting this option creates a new template:

<img width="613" alt="Screen Shot 2022-10-26 at 3 06 26 PM" src="https://user-images.githubusercontent.com/690791/198114643-ad50a7e2-a2b5-4653-8627-2e3bc171ca33.png">

This template is based upon the existing checkout UI extension with the following changes:

1. `shopify.ui.extension.toml` uses the new schema.
2. Two extension points are created (`Checkout::Dynamic::Render` and `Admin::Checkout::Editor`).
3. The readme has updated instructions to reflect that both of these extension points are created.

### How to test your changes?

1. Delete `fixtures/app/extensions`
2. Inside `fixtures/app` run `yarn generate extension` and select a pre-existing UI extension
3. Repeat step 2 for every UI extension EXCEPT the new one.
4. Run `yarn dev` to check that nothing broke the generation of existing extensions.
5. Run `yarn generate extension` and this time select the new UI Extension ( `Checkout UI Beta (shopifolk only)` ).

Creating a new extension should work. Note that if you run any command after creating the new extension it will fail because the CLI doesn't understand the new `toml` schema.

### Post-release steps

None

### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
